keps/sig-network/2112-netpol-namespace-name-selector/README.md

```diff
@@ -0,0 +1,393 @@
+we've seen in the broader community as well.
+
+We thus conclude that, even though targeting an object using its name is not 
+normal in K8s, in the case of NetworkPolicies, the overwhelming need for 
+universal, easy to define security boundaries, makes a strong case for 
+amending the API to support a "special" selector for namespace names that is 
+**independent** of labels.
+
+### Goals
+
+- Add a `namespaceNames` option (which is additive to the current `matchLabels` selector).
+- Allow multiple `namespaceNames` values, so that many namespaces can be 
+allowed using the same field 
+
+### Non-Goals
+
+- Support matching of wildcard namespaces
+- Changing the internal details of how namespaceSelector works with arbitrary syntax
+- Depending on more sophisticated features (like virtual labels) to accomplish 
+this feature without changing the API (note that, if someone were 
+to write a virtualLabels KEP, however, it might change the trajectory of this 
+KEP, which is worth discussing)
+
+### User Stories (Optional)
+
+See motivation section for user stories.  This KEP collects concrete data and opinions from several individuals over the past few years, hence we do not include generic user stories.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- NetworkPolicy providers may **opt-out**, initially to support this construct, and dilligent communication with CNI providers will be needed to make sure its widely adopted, thus, this feature needs to be backwards compatible with the existing v1 api.
+- We thus need to make sure that hidden defaults don't break the meaning of existing policys, for example:
+  - if `namespaceNames` field is retrieved by an older client which **doesn't** yet support it:
+    - the client doesnt crash (and the plugin doesnt crash, either).
+  - if `namespaceNames` is nil:
+    - the policy should behave identically to any policy made before this field was added.
+    - in other words, there is no "deny all" semantic that can be enforced by this namespace being missing.
+  - if `namespaceNames` is empty:
+    - it has IDENTICAL semantics as if it were nil.
+    - in other words, there is no "allow all" namespaces semantic which corresponds to emptyness
+
+Overall, the `NetworkPolicyPeer` modifications should be unobtrusive with regard to how other fields interplay, regardless of its absence or presence, notwithstanding the fact that it potentially broadens the selection of the `namespaceSelector` field.
+
+## Proposal
+
+In NetworkPolicy specification, inside `NetworkPolicyPeer` specify a new `namespaceNames` field.  
+
+1) One simple way to implement this feature is to modify the NetworkPolicyPeer 
+object, like so.  Since this doesn't involve modifying a go type (i.e. were 
+not replacing a labelSelector with a different type, we expect it 
+to be a cleaner implementation:
+
+```
+type NetworkPolicyPeer struct {
+  // new field
+  namespaceNames []string
```

Comment:
Probably per api design, this needs to be upper case first letter, right? (also because this field needs to be exported)